File: Python/Fibonacci SubProblems/nth_fibonacci_number.py
def nthfibonacci(n):
    a = [0 for _ in range(n+1)]
    a[0] = 0
    a[1] = 1
    for i in range(2, n+1):
        a[i] = a[i-1]+a[i-2]
    return a[n]

test = int(input())
for i in range(test):
    n = int(input())
    print(nthfibonacci(n))

File: C++/Fibonacci SubProblems/nth_fibonacci_number.cpp
#include<bits/stdc++.h>
using namespace std;

int nthfibonacci(int n){
    int a[n+1];
    for(int i = 0; i<n+1; i++){
        a[i] = 0;
    }
    a[1] = 1;

    for(int i = 2; i<n+1; i++){
       a[i] = a[i-1] + a[i-2];
    }
    return a[n];
}

int main(){
    int test, n;
    cin>>test;
    while(test--){
        cin>>n;
        cout<<nthfibonacci(n)<<'\n';
    }
}